sql_lineage/lineage.py

```diff
@@ -16,8 +18,12 @@ def _parse_ctes(sql: str) -> [Nodes, Edges]:
     """
     Parse an SQL query into nodes and edges based on CTEs.
     """
-    parsed = qualify.qualify(sqlglot.parse_one(sql))
     nodes, edges = [], []
+    parsed = sqlglot.parse_one(sql)
+    with contextlib.suppress(Exception):
+        # There isn't a hard requirement for the query to be valid SQL,
+        # but it's better if it is
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **question (code_clarification):** Clarify the comment about SQL validity.

The comment about SQL validity is a bit vague. It would be helpful to explain why it's beneficial for the query to be valid SQL, such as error handling or performance implications.